### PR TITLE
Phoenix interactive client

### PR DESCRIPTION
**Pull request by AleksandarFilipov**
Idea is that this is a client, thus it's added a  new  application in the umbrella. Changes were required in the base code which is the app `ae_socket_connector`. Adding applications is a way to test the base foundations for state channels.

Down the road another application will be added which suggests how an automatic backend can be implemented (already being worked on).
__________________________

Added Phoenix with a web interface which allows/show channels interaction.
Basic operations implemented. but more to come, this marked as WIP

Operator needs to select whether he's acting as initiator or responder. 

To run this on testnet:
`AE_NODE_URL="wss://testnet.aeternity.io:443/channel" AE_NODE_NETWORK_ID="ae_uat" iex -S mix phx.server`

Warning: both peers need to connect to same node, beware of load balancers.

This PR looks huge, however it's mostly due to Phoenix framework. Code changes are very limited in the original code `apps/ae_socket_connector`
Some js wiz should help and shape up `channel_interface/assets/js/app.js`

### CODE REVIEW

**Comment by tolbrino** [comment]
The PR needs to be rebased. Also the app name `AeChannelInterface` seems pretty undescriptive. How about `AeChannelInteractiveWebclient`?

**Comment by AleksandarFilipov** [comment]
> The PR needs to be rebased. Also the app name `AeChannelInterface` seems pretty undescriptive. How about `AeChannelInteractiveWebclient`?

Very valid, would prefer postponing this. Let do renaming as a seperate PR once master is up to speed.
PR is now rebased

**Comment by tolbrino** [comment]
> > The PR needs to be rebased. Also the app name `AeChannelInterface` seems pretty undescriptive. How about `AeChannelInteractiveWebclient`?
> 
> Very valid, would prefer postponing this. Let do renaming as a seperate PR once master is up to speed.
> PR is now rebased

Postponing is always a recipe for forgetting it. Can you create a ticket so it stays somewhere? Also tests are failing.

**Comment by AleksandarFilipov** [comment]
It's now green to go. 

https://github.com/aeternity/ae-channel-service/issues/52

File: apps/ae_channel_interface/config/config.exs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+# This configuration file is loaded before any dependency and
+# is restricted to this project.
+
+# General application configuration
+use Mix.Config
+
+# Configures the endpoint
+config :ae_channel_interface, AeChannelInterfaceWeb.Endpoint,
+  url: [host: "localhost"],
+  secret_key_base: "KmiLveFrbi3Zn4MCPdXza9hu+I38jI47ql/jasWqa3pWJTWm0RgHtI+JOG6I9t/l",
```
Comment (tolbrino) [inline apps/ae_channel_interface/config/config.exs:13]: Please remove secret from version control. Should be set via shell env var.

File: apps/ae_channel_interface/assets/package.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
+  "repository": {},
+  "license": "MIT",
```
Comment (tolbrino) [inline apps/ae_channel_interface/assets/package.json]: Should be ISC.

File: apps/ae_channel_interface/README.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# AeChannelInterface
```
Comment (tolbrino) [inline apps/ae_channel_interface/README.md:1]: Please remove placeholder file.

File: apps/ae_channel_interface/lib/ae_channel_interface_web/endpoint.ex [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  plug Plug.MethodOverride
+  plug Plug.Head
+
+  # The session will be stored in the cookie and signed,
+  # this means its contents can be read but not tampered with.
+  # Set :encryption_salt if you would also like to encrypt it.
+  plug Plug.Session,
+    store: :cookie,
+    key: "_ae_channel_interface_key",
+    signing_salt: "37O4Mnmk"
```
Comment (tolbrino) [inline apps/ae_channel_interface/lib/ae_channel_interface_web/endpoint.ex:43]: Make salt configurable per environment and secret.

File: apps/ae_channel_interface/lib/ae_channel_interface_web/endpoint.ex [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+  plug Plug.MethodOverride
+  plug Plug.Head
+
+  # The session will be stored in the cookie and signed,
+  # this means its contents can be read but not tampered with.
+  # Set :encryption_salt if you would also like to encrypt it.
+  plug Plug.Session,
+    store: :cookie,
+    key: "_ae_channel_interface_key",
```
Comment (tolbrino) [inline apps/ae_channel_interface/lib/ae_channel_interface_web/endpoint.ex:42]: Make key configurable per environment.

**Review by tolbrino — APPROVED** [review]
Please address comment in follow-up PR.

File: apps/ae_channel_interface/assets/package.json (hunk 2 above)
Comment (AleksandarFilipov) [reply apps/ae_channel_interface/assets/package.json]: Fixed

**Pull request merged**